### PR TITLE
replacestring and getaccountname

### PR DESCRIPTION
**Pull request by Jan-Kruis**
Replace _String will replace <date> with real date from the vacation table

get_account will get name from mailbox a account name to place as friendly name to send as sender's email

### CODE REVIEW

File: VIRTUAL_VACATION/vacation.pl [hunk 1]
```diff
@@ -332,9 +356,67 @@ sub check_for_vacation {
...
+    my $query = qq{SELECT body,activefrom,activeuntil FROM vacation WHERE email=? };
+    my $stm = $dbh->prepare($query) or panic_prepare($query);
+    $stm->execute($to) or panic_execute($query,"email='$to'");
+    my @row = $stm->fetchrow_array;
+    my $rv = $stm->rows;
 
-# try and determine if email address has vacation turned on; we
-# have to do alias searching, and domain aliasing resolution for this.
+    my $vacation_body = $row[0];
+    my $f_date = substr($row[1],0,10);
```
Comment (DavidGoodwin) [inline VIRTUAL_VACATION/vacation.pl:394]: why do you need substr calls here?
row[1] should be a database timestamp field, shouldn't it?

**Comment by DavidGoodwin** [comment]
Thanks for the PR - see comment above.

**Comment by Jan-Kruis** [comment]
Hi David, you are right when I started I had problems with the conversion after the date format. the solution substr worked, should not get the time message and no more thought to adjust it. 

I came across another typo 

New code


	my $vacation_body = $row[0]; 
+->	my $f_date = $row[1]; 
+->	my $u_date = $row[2]; 

	# #Note!! do not replace '%Y-%m-%d' with date_format because this is the format that f_date and u_date are filled with date in this format 
	# $date_formart is used to display the dates in your choice of format. 
	# 

	my $date_f = Time::Piece->strptime($f_date,'%Y-%m-%d’);
	$f_date = $date_f->strftime($date_format); 
	my $date_u = Time::Piece->strptime($u_date,'%Y-%m-%d’);
+->	$u_date = $date_u->strftime($date_format);   <<== was('$date_format’)

I send a new pull request

File: VIRTUAL_VACATION/Contributions.txt [hunk 2]
```diff
@@ -91,3 +91,9 @@ Additional authors:
             sending vacation mails, even if one or multiple of the recipients the alias points to has vacation
             currently active. 
 
+2022-10-20  Jan Kruis <jan@crossreference.nl>
+            Add configuration parameter $replace_from ,$replace_until and $date_format for the subroutine replace_string
+            the subroutine repalce_string replaces in the bodytext the text defined at $replace_from and $replace_until with the date of active from and active until in the format specified at $date_format.
```
Comment (cboltz) [inline VIRTUAL_VACATION/Contributions.txt:96]: Typo: `subroutine rep*la*ce_string`

File: VIRTUAL_VACATION/Contributions.txt [hunk 3]
```diff
@@ -91,3 +91,9 @@ Additional authors:
...
+
+	    Add configuration parameter $account_check and $account_name if account_check is set it wil add de value of name of the mailbox in front of de email and it is shown as sender,
```
Comment (cboltz) [inline VIRTUAL_VACATION/Contributions.txt:98]: Typos: `it wil*l* add *the*` and `of *the* email`

File: VIRTUAL_VACATION/vacation.pl [hunk 4]
```diff
@@ -124,6 +133,21 @@
 # our $no_vacation_pattern = 'info\@example\.org';
 our $no_vacation_pattern = 'info\@example\.org';
 
+#
+# The subroutine repalce_string replaces in the body text the text defined at $replace_from and $replace_until with the date of activefrom and activeunitul in the format specified at $date_format.
```
Comment (cboltz) [inline VIRTUAL_VACATION/vacation.pl:137]: Another `rep*la*ce_string` typo

File: VIRTUAL_VACATION/vacation.pl [hunk 5]
```diff
@@ -332,9 +356,67 @@ sub check_for_vacation {
...
+    my $accountname = $row[0];
+
+    return $accountname;
+}
+
+#
+# Replace <%From_Date>   with date part of activefrom out the vacation table on base of email
+# Replace <%Until_Date>  with date part of activeuntil out the vacation table on base of email
+#
+# The varibale $replace_form and $replace_until will have the <%From_Date> and <%Until_Date> replace text
```
Comment (cboltz) [inline VIRTUAL_VACATION/vacation.pl:381]: `replace_f*ro*m`

File: VIRTUAL_VACATION/vacation.pl [hunk 6]
```diff
@@ -332,9 +356,67 @@ sub check_for_vacation {
...
+    my $rv = $stm->rows;
 
-# try and determine if email address has vacation turned on; we
-# have to do alias searching, and domain aliasing resolution for this.
+    my $vacation_body = $row[0];
+    my $f_date = $row[1];
+    my $u_date = $row[2];
+#
+# Note !!  do not  replace '%Y-%m-%d' with date_format because this is the format that f_date and u_date are are filled with date in this format
+# $date_formart is used to diplay the dates in your choice of format.
```
Comment (cboltz) [inline VIRTUAL_VACATION/vacation.pl:398]: `$date_for*ma*t`

**Comment by Jan-Kruis** [comment]
hi christian,

thanks for correcting my typos, i suffer from dyslexia so read about my typos often and swap letters too

File: VIRTUAL_VACATION/Contributions.txt (hunk 2 above)
Comment (Jan-Kruis) [reply VIRTUAL_VACATION/Contributions.txt]: i changed the typo error's

File: VIRTUAL_VACATION/vacation.pl (hunk 4 above)
Comment (Jan-Kruis) [reply VIRTUAL_VACATION/vacation.pl]: i changed the typo error's

File: VIRTUAL_VACATION/vacation.pl (hunk 5 above)
Comment (Jan-Kruis) [reply VIRTUAL_VACATION/vacation.pl]: i changed the typo error's

File: VIRTUAL_VACATION/vacation.pl (hunk 6 above)
Comment (Jan-Kruis) [reply VIRTUAL_VACATION/vacation.pl]: i changed the typo error's

**Pull request merged**